perf_test/blas/blas3/KokkosBlas3_trmm_perf_test.hpp

```diff
@@ -72,6 +72,26 @@ void (*do_trmm_invoke[LOOP_N][TEST_N])(options_t) = {
 #define DEFAULT_TRMM_ARGS "LUNU"
 #define DEFAULT_TRMM_ALPHA 1.0
 
+/**
+ * The KokkosBatched::SerialTrmm implementation performs dot products on
+ * non-zero elements of the triangular matrices. The flop calculation below
+ * assumes KokkosBatched::SerialTrmm is being used. Since the dot products
+ * do a multiply and add we can calculate the flops for any element in the last
+ * column of the LHS to be 2*columns_LHS, any element in the last-1 column of
+ * the LHS to be 2*(columns_LHS-1), and so on. We do this for every row of the LHS
+ * giving us this flop count:
+ *  flops = columns_LHS * (columns_LHS + 1)
+ *  flops = (flops / 2) * 2
+ *  flops = flops * rows_LHS
+ */
+static inline int trmm_flop_count(char side, int b_m, int b_n, int a_m, int a_n) {
+  if (side == 'L' || side == 'l') {
+    return (a_n * (a_n + 1)) * a_m;
+  } else {
+    return (b_n * (b_n + 1)) * b_m;
+  }
+}
+
```

Comment:
- I was not sure why we have +1 here.. Shouldn't it be b_n*b_n*b_m for side == right?
- For side == left, shouldn't it be b_n*b_m*b_m (e.g., A is square and a_m==a_n)?
- Does the tester support only real values? For complex, shouldn't we multiply the flop count for the addition and multiplication by two and six?

Reply:
>     * I was not sure why we have +1 here

I used a gauss sum to count flops for each row of B and multiplied that by 2 (since there is one addition and one subtraction). Then I multiplied this by the number of rows in B.

>     * For side == left, shouldn't it be b_n_b_m_b_m

We use a_n and a_m for side == left.

>     * Does the tester support only real values? For complex, shouldn't we multiply the flop count for the addition and multiplication by two and six?

I will update the flop counter to account for complex numbers.

Reply:
> `I used a gauss sum to count flops for each row of B and multiplied that by 2 (since there is one addition and one subtraction). Then I multiplied this by the number of rows in B.`

So, these are the numbers of flops the implementation performs, and not the "paper-and-pencil" flop count, which needs M * N less additions?

> 'We use a_n and a_m for side == left.'

Right. And since A is square, you are computing it as M^3? Shouldn't that be M^2 * N?